beluga-aws-sdk/src/lib.rs

```diff
@@ -1,14 +1,17 @@
-use beluga_mqtt::{MqttClient, QoS, Subscriber};
+use beluga_mqtt::{MqttClient, Publish, QoS};
 use beluga_tunnel::Tunnel;
 use error::Error;
-use tokio::task::JoinHandle;
+use tokio::task::{JoinHandle, JoinSet};
+use tokio_util::sync::{CancellationToken, DropGuard};
+use tracing::debug;
 
 mod error;
 
 type Result<T> = std::result::Result<T, Error>;
 
 pub struct TunnelManager {
-    subscriber: Subscriber,
+    _handler: JoinHandle<Result<()>>,
+    _guard: DropGuard,
 }
 
 impl TunnelManager {
```

Comment:
We're still missing the `async shutdown` function.
The shutdown function should:

1. cancel the cancellation token
2. wait for all the tasks in the join set to complete (by repeatedly calling `join_next`).